src/main/java/io/fabric8/elasticsearch/cloud/kubernetes/KubernetesAPIServiceImpl.java

```diff
@@ -28,20 +32,29 @@
 
   private final String namespace;
   private final String serviceName;
+  private final String podLabel;
 
   @Override
   public Endpoints endpoints() {
     logger.debug("get endpoints for service {}, namespace {}", serviceName, namespace);
     return client().endpoints().inNamespace(namespace).withName(serviceName).get();
   }
 
+  @Override
+  public List<Pod> pods() {
+    logger.debug("get endpoints with pod label {}, namespace {}", podLabel, namespace);
+    final String[] l = podLabel.split("=");
+    return client().pods().inNamespace(namespace).withLabel(l[0], l[1]).list().getItems();
```

Comment:
The pod label will be defined [here](https://github.com/openshift/openshift-ansible/blob/46d9ba942e49e4b6953dc8fa47e38d6ea98ec122/roles/openshift_logging_elasticsearch/templates/elasticsearch.yml.j2#L26). The label selector API seems to be very flexible allowing us to implement pretty much any type of selection we would like, the above algorithm was the simplest.

Should I allow selection based on the key only, or multiple labels?